pingcap/tidb
# metrics: add different labels for restricted SQL and general SQL

The newly added label is `sql_type`, it has value: `general` and `internal`,
users can use this label to filter metrics they have interest.

Also, add keyword `[INTERNAL]` for restricted SQL printed in slow log for
convenient identification.



### What problem does this PR solve? 

Improve usability of TiDB. We output bunch of metrics for monitoring status of TiDB now, such as  time spent on distsql. Sometimes users may notice high distsql time on dashboard, and they may wonder why my simple SQL is spending much time here, but actually, the monitored distsql time may belong to internal SQL queries(called restricted SQL) issued by DDL or auto Analyze for example. To avoid this confusion, we differentiate restricted SQL and general SQL in metrics, users can choose to only reflect metrics of general SQL on dashboard now, or they can monitor restricted SQL status as well.


### What is changed and how it works?

add label for:
- `DistSQLQueryHistgram`
- `QueryDurationHistogram`
- `SessionExecuteParseDuration`
- `SessionExecuteCompileDuration`
- `SessionExecuteRunDuration`

Also, add keyword `[INTERNAL]` for restricted SQL printed in slow log for convenient identification.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

    deploy a TiDB instance with pushgateway, prometheus and grafana for monitor(see also https://github.com/pingcap/tidb-ansible for instructions), run some workload on TiDB instance, then monitor status in grafana using the newly added label `sql_type` for filtering.

Related changes

 - Need to be included in the release note: `add different labels for restricted SQL and general SQL in metrics`
